src/query/service/src/interpreters/interpreter_show_columns.rs

```diff
@@ -0,0 +1,258 @@
+            } else {
+                return Err(ErrorCode::Internal(
+                    "Logical error, View Table must have a SelectQuery inside.",
+                ));
+            }
+        } else {
+            table.schema().fields().clone()
+        };
+
+        let fields_len = fields.len();
+        let mut names: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut data_types: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut is_nullables: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut exprs: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut extras: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut keys: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        // full
+        let mut collations: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut privileges: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut comments: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+
+        // Append columns.
+        {
+            for field in fields.into_iter() {
+                names.push(field.name().clone().into_bytes());
+                let data_type = field.data_type().remove_recursive_nullable().sql_name();
+                data_types.push(data_type.into_bytes());
+
+                let mut default_expr = "".to_string();
+                if let Some(expr) = field.default_expr() {
+                    default_expr = expr.to_string();
+                }
+                exprs.push(default_expr.into_bytes());
+                if field.is_nullable() {
+                    is_nullables.push("YES".to_string().into_bytes());
+                } else {
+                    is_nullables.push("NO".to_string().into_bytes());
+                }
+                extras.push("".to_string().into_bytes());
+                keys.push("".to_string().into_bytes());
+                if self.plan.full {
+                    collations.push("".to_string().into_bytes());
+                    privileges.push("".to_string().into_bytes());
+                    comments.push("".to_string().into_bytes());
+                }
+            }
+        }
+
+        let block = if self.plan.full {
+            DataBlock::new_from_columns(vec![
+                // | Field | Type | Collation | Null | Key  | Default | Extra | Privileges | Comment |
+                StringType::from_data(names),
+                StringType::from_data(data_types),
+                StringType::from_data(collations),
+                StringType::from_data(is_nullables),
+                StringType::from_data(keys),
+                StringType::from_data(exprs),
+                StringType::from_data(extras),
+                StringType::from_data(privileges),
+                StringType::from_data(comments),
+            ])
+        } else {
+            // | Field | Type | Null | Key  | Default | Extra |
+            DataBlock::new_from_columns(vec![
+                StringType::from_data(names),
```

Comment:
It seems to have some repetitive code that could be simplified.

Reply:
I think we can not do. Need ensure that the sequence is the same as the schema